source/gravity_model/ascii_data.cc

```diff
@@ -0,0 +1,118 @@
+/*
+  Copyright (C) 2011 - 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#include <aspect/gravity_model/ascii_data.h>
+
+#include <aspect/geometry_model/box.h>
+#include <aspect/geometry_model/spherical_shell.h>
+#include <aspect/geometry_model/sphere.h>
+#include <aspect/geometry_model/chunk.h>
+#include <aspect/geometry_model/ellipsoidal_chunk.h>
+
+namespace aspect
+{
+  namespace GravityModel
+  {
+    template <int dim>
+    AsciiData<dim>::AsciiData ()
+    {}
+
+
+    template <int dim>
+    void
+    AsciiData<dim>::initialize ()
+    {
+      this->initialize(7,this->get_mpi_communicator());
+    }
+
+
+    template <int dim>
+    Tensor<1,dim>
+    AsciiData<dim>::
+    gravity_vector (const Point<dim> &position) const
+    {
+      const double depth = this->get_geometry_model().depth(position);
+      const double magnitude = this->get_data_component(Point<1>(depth),3);
+      Point<dim> vertical_direction;
+
+      // in dependence of what the geometry model is, gravity points in a different direction
+      if (const GeometryModel::SphericalShell<dim> *geometry = dynamic_cast<const GeometryModel::SphericalShell<dim>*> (&this->get_geometry_model()))
```

Comment:
I wonder if there is a better way to do this. Maybe we can introduce a property that provides a "downward direction" for a point x? At least you can write this as something like
```
if (dynamic_cast<const GeometryModel::SphericalShell<dim>*> (&this->get_geometry_model())
|| dynamic_cast<const GeometryModel::Chunk<dim>*> (&this->get_geometry_model())
|| ... )
return - magnitude * position/position.norm();
else
...
```
(note that you don't use the ``gm`` variable)